autovalet_gazebo/scripts/autovalet_system.py

```diff
@@ -0,0 +1,176 @@
+        output = "Current State: "
+
+        if self.plan_state == State.Waiting:
+            output += "WAITING FOR NEXT GOAL"
+        elif self.plan_state == State.Planning:
+            output += "PLANNING PATH TO GOAL"
+        elif self.plan_state == State.Executing:
+            output += "EXECUTING PLAN"
+        elif self.plan_state == State.Fail:
+            output += "FAILED TO FIND A PLAN"
+        elif self.plan_state == State.Success:
+            output += "SUCCESSFULLY REACHED GOAL"
+
+        rospy.logwarn(output)
+
+    def goalCallback(self,goal_msg):
+        
+
+        self.plan_state = State.Planning
+
+        self.goal_pose = goal_msg
+
+        self.start_pose = getCurrentPose()
+
+        self.printState()
+
+        rospy.logwarn("Goal:")
+        rospy.logwarn("Position")
+        rospy.logwarn("  x: %f",self.goal_pose.pose.position.x)
+        rospy.logwarn("  y: %f",self.goal_pose.pose.position.y)
+        rospy.logwarn("  z: %f",self.goal_pose.pose.position.z)
+        rospy.logwarn("Orientation");
+        rospy.logwarn("  x: %f",self.goal_pose.pose.orientation.x)
+        rospy.logwarn("  y: %f",self.goal_pose.pose.orientation.y)
+        rospy.logwarn("  z: %f",self.goal_pose.pose.orientation.z)
+        rospy.logwarn("  w: %f",self.goal_pose.pose.orientation.w)
+
+    def MBstatusCallback(self,status_array_msg):
+
+        # print(status_array_msg.__dict__)
+        ACTIVE = 1
+        SUCCEEDED = 3
+        ABORTED = 4
+
+        if status_array_msg.status_list:
+            current_status = status_array_msg.status_list[-1]
+
+            if current_status.status != ACTIVE and self.plan_state == State.Waiting:
+                return
+
+            if current_status.status == ACTIVE:
+                if self.plan_state == State.Waiting:
+                    self.plan_state = State.Planning
+            
+                elif self.plan_state == State.Planning:
+                    current_pose = getCurrentPose()
+
+                    if not arePosesClose(current_pose,self.start_pose):
+                        self.plan_state = State.Executing
+                    else:
+                        return
+                elif self.plan_state == State.Executing:
+                    return
+
+            elif current_status.status == SUCCEEDED:
+                if self.plan_state == State.Waiting:
+                    return
+                else:
+                    self.plan_state = State.Success
+                    self.printState()
+                    self.plan_state = State.Waiting
+            
+            elif current_status.status == ABORTED and self.plan_state != State.Waiting:
+                if self.plan_state == State.Waiting:
+                    return
+
+                else:
+                    self.plan_state = State.Fail
+                    self.printState()
+                    self.plan_state = State.Waiting
+
+        else:
+            return
+
+        self.printState()
```

Comment:
Since there will be more classes like this one, should we separate each subsystem class into different files and import them here? just lmk if I'm making sense. We can change this in the next MR.